diff --git a/dist/esm/ReactHookForm.js b/dist/esm/ReactHookForm.js
index d0a19964fbee873424246aec5bcc86f2c95823da..cbd767daab74a21e75d7623a7809cf7dce24d9ff 100644
--- a/dist/esm/ReactHookForm.js
+++ b/dist/esm/ReactHookForm.js
@@ -199,11 +199,7 @@ export const layer = Button => Layer.effect(FormFramework.FormFramework, Effect.
         const formMethods = RHF.useForm({
           mode: props.validationMode ?? "onBlur",
           resolver: effectTsResolver(schema),
-          defaultValues: resetValues_,
-          values: initialValues,
-          resetOptions: {
-            keepDefaultValues: true
-          }
+          defaultValues: initialValues ?? resetValues_,
         });
         return React.createElement(RHF.FormProvider, {
           ...formMethods
@@ -212,7 +208,7 @@ export const layer = Button => Layer.effect(FormFramework.FormFramework, Effect.
           onSubmit: formMethods.handleSubmit(values => onSubmit({
             decoded: values,
             encoded: Schema.encodeUnknownSync(schema)(values)
-          }), onError)
+          }, formMethods), onError)
         }, children));
       };
       return Object.assign(Form, {
